src/templates/antora/util.ts

```diff
@@ -0,0 +1,270 @@
+      const fullName = type.fullName.toLowerCase();
+      return getNamespaceFromFullName(fullName) === url;
+    }).map((type): NavFile => {
+      return { title: type.fullName, path: type.fullName.toLowerCase() };
+    });
+
+    return {
+      title,
+      path: url,
+      pages: innerPages
+    };
+  });
+
+  return {
+    title: 'API Reference',
+    path: BASE_PATH,
+    pages
+  };
+};
+
+const navToAdoc = (indexPage: NavFile, structure: ExportStructure): string => {
+  // Api index nav page top level
+  let adoc = navLine(indexPage.title, 1);
+  if (indexPage.pages) {
+    indexPage.pages.forEach((namespace) => {
+      adoc += navLine(namespaceNavLine(namespace, structure), 2);
+      if (namespace.pages) {
+        namespace.pages.forEach((pageFile) => {
+          adoc += navLine(pageFileNavLine(pageFile, structure), 3);
+        });
+      }
+    });
+  }
+  return adoc;
+};
+
+const generateNavPages = (indexPage: NavFile, structure: ExportStructure): PageOutput[] => {
+  const navPages = [] as PageOutput[];
+  navPages.push({
+    type: 'adoc',
+    filename: '_data/nav.yml',
+    content: YAML.dump(indexPage)
+  });
+
+  const adocNav = navToAdoc(indexPage, structure);
+  navPages.push({
+    type: 'adoc',
+    filename: '_data/moxiedoc_nav.adoc',
+    content: adocNav
+  });
+  return navPages;
+};
+
+const legacyIndexToAdoc = (
+  namespace: NavFile,
+  template: HandlebarsTemplateDelegate,
+  descriptions: Record<string, string>,
+  structure: ExportStructure
+): string => {
+  const keywords = [ getApiFromFullName(namespace.title) ];
+  const indexPageLines = [
+    '\n== ' + namespaceDescriptions[namespace.title.toLowerCase()] + '\n\n'
+  ];
+  if (namespace.pages) {
+    indexPageLines.push(
+      '[cols="1,1"]\n',
+      '|===\n\n'
+    );
+    namespace.pages.forEach((pageFile) => {
+      keywords.push(getNameFromFullName(pageFile.title));
+      indexPageLines.push('a|\n');
+      indexPageLines.push('[.lead]\n');
+      indexPageLines.push(pageFileLegacyIndexLine(pageFile, structure) + '\n\n');
+      const description = descriptions[pageFile.path];
+      indexPageLines.push(description + '\n\n');
+    });
+    if (namespace.pages.length % 2 !== 0) {
+      indexPageLines.push('a|\n');
+    }
+    indexPageLines.push('|===\n');
+  }
+  const data = {
+    fullName: namespace.title,
+    desc: namespaceDescriptions[namespace.title.toLowerCase()],
+    keywords: keywords.join(', ')
+  };
+  const adoc = template(data) + indexPageLines.join('');
+  return adoc;
+};
+
+const generateLegacyIndexPages = (
+  indexPage: NavFile,
+  sortedTypes: Type[],
+  memberTemplate: HandlebarsTemplateDelegate,
+  structure: ExportStructure
+): PageOutput[] => {
+  const newNavPages: PageOutput[] = [];
+  const descriptions = getDescriptionsFromTypes(sortedTypes);
+  indexPage.pages.forEach((namespace) =>
+    newNavPages.push({
+      type: 'adoc',
+      filename: BASE_PATH + '/api/' + namespace.path + '/index.adoc',
```

Comment:
This could also be a spot where double slashes are added unsure.

Reply:
`namespace.path` shouldn't ever have a slash, only `.` separators for the apis, and BASE_PATH shouldn't ever end in a slash.

Reply:
Since it's user specified it could. But I guess we are smarter than that right. :)